### PR TITLE
Support multiple networks

### PR DESCRIPTION
**Pull request by baude**
This is a refresh of Dan William's PR #974 with a rebase and proper
vendoring of ocicni and containernetworking/cni.  It adds the ability
to define multiple networks as so:

podman run --network=net1,net2,foobar ...

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: libpod/container.go [hunk 1]
```diff
@@ -146,15 +146,11 @@ type containerState struct {
 	// IPs contains IP addresses assigned to the container
```
Comment (mheon) [inline libpod/container.go]: Trim this comment, we killed the IP field

File: libpod/container.go [hunk 2]
```diff
@@ -146,15 +146,11 @@ type containerState struct {
...
-	Interfaces []*cnitypes.Interface `json:"interfaces,omitempty"`
-	// Routes contains network routes present in the container
-	// Only populated if we created a network namespace for the container,
-	// and the network namespace is currently active
-	Routes []*types.Route `json:"routes,omitempty"`
+	// NetworkResults contains the configuration results for all networks
+	// the pod is attached to. Only populated if we created a network
+	// namespace for the container, and the network namespace is currently
+	// active
+	NetworkResults []*cnitypes.Result `json:"networkResults,omitempty"`
```
Comment (mheon) [inline libpod/container.go]: Rename to just Networks I think

File: libpod/container.go [hunk 3]
```diff
@@ -268,6 +264,8 @@ type ContainerConfig struct {
 	// Hosts to add in container
 	// Will be appended to host's host file
 	HostAdd []string `json:"hostsAdd,omitempty"`
+	// Network names to add container to. Empty to use default network.
```
Comment (mheon) [inline libpod/container.go]: CNI Network names, not Network Names

File: libpod/container.go (hunk 2 above)
Comment (mheon) [reply libpod/container.go]: or NetworkStatus. That would be better

File: libpod/runtime.go [hunk 4]
```diff
@@ -133,7 +133,10 @@ type RuntimeConfig struct {
 	// plugins can be located
 	CNIPluginDir []string `toml:"cni_plugin_dir"`
 	// HooksDir Path to the directory containing hooks configuration files
-	HooksDir string `toml:"hooks_dir"`
+	// CNIDefaultNetwork is the network name of the default CNI network
+	// to attach pods to
+	CNIDefaultNetwork string `toml:"cni_default_network,omitempty"`
+	HooksDir          string `toml:"hooks_dir"`
```
Comment (mheon) [inline libpod/runtime.go]: Comments messed up here as well - HooksDir comment got lost above

File: vendor.conf [hunk 5]
```diff
@@ -8,7 +8,7 @@ github.com/boltdb/bolt master
 github.com/buger/goterm 2f8dfbc7dbbff5dd1d391ed91482c24df243b2d3
 github.com/containerd/cgroups 77e628511d924b13a77cebdc73b757a47f6d751b
 github.com/containerd/continuity master
-github.com/containernetworking/cni v0.6.0
+github.com/containernetworking/cni ffc1e7b446d795f885311edf393c6209c32855c6
```
Comment (mheon) [inline vendor.conf]: Can we get a tag here, instead of a straight commit? Is there a tag we can use?

**Comment by mheon** [comment]
I'd like an OK from one of the CNI maintainers that the commit we're pinning is sufficiently stable to use before we merge

File: libpod/container.go (hunk 2 above)
Comment (dcbw) [reply libpod/container.go]: +1 for NetworkStatus

**Comment by dcbw** [comment]
> I'd like an OK from one of the CNI maintainers that the commit we're pinning is sufficiently stable to use before we merge

@mheon Yes, it is sufficiently stable as long as you're not using any of the GET stuff, which I don't believe podman/ocicni are.

File: vendor.conf (hunk 5 above)
Comment (baude) [reply vendor.conf]: no

File: libpod/container.go (hunk 1 above)
Comment (baude) [reply libpod/container.go]: done

File: libpod/container.go (hunk 2 above)
Comment (baude) [reply libpod/container.go]: done

File: libpod/container.go (hunk 3 above)
Comment (baude) [reply libpod/container.go]: fixed

File: libpod/runtime.go (hunk 4 above)
Comment (baude) [reply libpod/runtime.go]: fixed

File: vendor.conf (hunk 5 above)
Comment (baude) [reply vendor.conf]: ok fixed

**Comment by baude** [comment]
@rhatdan @mheon merge ?

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by baude** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**